src/ray/object_manager/plasma/store.cc

```diff
@@ -190,76 +188,16 @@ void PlasmaStore::Stop() { acceptor_.close(); }
 // If this client is not already using the object, add the client to the
 // object's list of clients, otherwise do nothing.
 void PlasmaStore::AddToClientObjectIds(const ObjectID &object_id,
-                                       const LocalObject *entry,
                                        const std::shared_ptr<Client> &client) {
   // Check if this client is already using the object.
   if (client->object_ids.find(object_id) != client->object_ids.end()) {
     return;
   }
-  // If there are no other clients using this object, notify the eviction policy
-  // that the object is being used.
-  if (entry->ref_count == 0) {
-    // Tell the eviction policy that this object is being used.
-    eviction_policy_.BeginObjectAccess(object_id);
-    num_bytes_in_use_ += entry->GetObjectSize();
-  }
-  // Increase reference count.
-  entry->ref_count++;
-  RAY_LOG(DEBUG) << "Object " << object_id << " in use by client"
-                 << ", num bytes in use is now " << num_bytes_in_use_;
-
+  object_lifecycle_mgr_.AddReference(object_id);
```

Comment:
Given we said internal invariant belongs to the class, should we have a check here? (I think previously, we had this check because entry is always not a nullptr).`(RAY_CHECK(object_lifecycle_mgr_.AddReference(object_id));` or maybe check if object exists in local before adding a reference?